nrf_rpc/nrf_rpc.c

```diff
@@ -760,27 +761,44 @@ int nrf_rpc_rsp_send(nrf_rpc_alloc_ctx ctx, uint8_t *packet, size_t len)
 
 int nrf_rpc_init(void)
 {
+	void *iter;
+	const struct nrf_rpc_group *group;
+	uint8_t group_id = 0;
+
 	NRF_RPC_DBG("Initializing nRF RPC Module");
+
+	for (NRF_RPC_AUTO_ARR_FOR(iter, group, &nrf_rpc_groups_array,
+				 const struct nrf_rpc_group)) {
+		
+		if (group_id > 0x7F) {
+			return -ENOMEM;
+		}
+		*group->group_id = group_id;
+		group_id++;
+	}
```

Comment:
I can add checking of groups to `INIT` packet that I added to changes that I work on now. I think that case with not matching groups will be not so common, so maybe just a checksum of a list of groups will be enough.

Reply:
I added `INIT` packet that has group count and a simple checksum (not CRC) of all group names. I think that it will be enough, because group mismatch will not be something that will happen normally.